### DISCUSSION TITLE
Anyone know how to configure the blacklist?

### DISCUSSION THREAD

**Discussion by GerryOnGithub** [tobychui/zoraxy#171] [General]
I installed Zoraxy yesterday and seems very nice. I have a huge list of blacklisted IP addresses (like 25K) configured in nginx but I don't understand how to bring them into Zoraxy.

I found this image which I guess points me to where I need to look (https://res.cloudinary.com/practicaldev/image/fetch/s--hYPShvlx--/c_limit%2Cf_auto%2Cfl_progressive%2Cq_auto%2Cw_800/https://dev-to-uploads.s3.amazonaws.com/uploads/articles/ddxqp4muzz0k3q2gul2r.png)

also I see there is a blacklist.html file that gets displayed, but it seems I don't understand how to blacklist IPs, any hints? I'm I not seeing some docs somewhere?

Later today: I see that image under blacklist isn't about blacklisting at all.

**Comment by tobychui** [comment]
Zoraxy is not based on nginx, so there are no way for you to easily move blacklisted IP into Zoraxy from nginx. But if you need a config file based migration, you can do that with the following steps.

1. Go to Access Rule Editor, create a new access rule
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/ae889cd1-9f6d-4e21-8682-8f7d6fd800dd)
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/93f744bb-81d0-4ac8-82ea-fd756a1e21b9)

2. Go to the `conf/access/` folder. You will see a few json file with random UUID. Pick the one you just created and open it with text editor. You will see something like this.
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/b194650a-ba95-4cba-8802-27ebf203fef7)

3. Add your IP into the JSON and set BlacklistEnabled to true. The key field can be IP address or CIDR. Here is an example
```json
{
 "ID": "6224f268-3cef-41ca-a34b-2739112edb34",
 "Name": "MyRuleWithTonsOfIps",
 "Desc": "Something important here",
 "BlacklistEnabled": true, //Set this to true
 "WhitelistEnabled": false,
 "WhiteListCountryCode": {},
 "WhiteListIP": {},
 "BlackListContryCode": {},
 "BlackListIP": {
    "127.0.0.0/24": "I dont like loopback ips"
//Add more ip here
 }
}
```
4. Restart Zoraxy
5.  Go to HTTP Proxy, click on the edit button of the rule you want to apply the access policy, and click "Access Rule"
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/065f0747-eca8-4e7d-9a54-d056f813a0bd)

6. Select the access rules you want to apply and click "Apply Change"
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/b3367cd8-cc28-481a-8c85-e59dcf476119)

**Reply by GerryOnGithub** [reply]
Thanks super helpful! Got lost at "Go to HTTP Proxy" because I have one domain thus use Default which doesn't have anything about rules but I'll keep looking

**Reply by tobychui** [reply]
![圖片](https://github.com/tobychui/zoraxy/assets/24617523/34e07021-3aff-4cf6-904a-155c709de4dc)
The HTTP Proxy tab on the left menu bar. It will also works if you put all blacklisted IP into default if you only have a single domain.
